src/machine.ts

```diff
@@ -0,0 +1,66 @@
+import { Path, PathType } from './path';
+
+export class State {
+  x: number;
+  y: number;
+  z: number;
+  e: number;
+  tool: number;
+  units: 'mm' | 'in';
+
+  static get initial(): State {
+    const state = new State();
+    Object.assign(state, { x: 0, y: 0, z: 0, e: 0, tool: 0, units: 'mm' });
+    return state;
+  }
+}
+
+export class Machine {
```

Comment:
It used to be called `Print`, but the name does not apply to the CNC case. 

I'm not sure about the `Machine` name but it's the best I could come up with so far. It is supposed to represent the state and result after running some GCode.

Reply:
`Job` is fantastic. Yes!